deployment/terraform/utils.go

```diff
@@ -352,20 +355,114 @@ func getServerURL(output *Output, deploymentConfig *deployment.Config) string {
 	return deploymentConfig.ServerScheme + "://" + url
 }
 
-// GetAWSConfig returns the AWS config, using the profile configured in the
-// deployer if present, and defaulting to the default credential chain otherwise
-func (t *Terraform) GetAWSConfig() (aws.Config, error) {
+// Durations for the expiry of the AWS Role credentials and for its refresh
+// interval, which, out of an abundance of caution, is a bit shorter than half
+// the expiry duration of the role.
+const (
+	// One hour is the maximum allowed by role chaining
+	AWSRoleTokenDuration        = 1 * time.Hour
+	AWSRoleTokenRefreshInterval = 25 * time.Minute
+)
+
+func (t *Terraform) InitCreds() error {
 	regionOpt := awsconfig.WithRegion(t.config.AWSRegion)
+	var opts []func(*awsconfig.LoadOptions) error
+
+	opts = append(opts, regionOpt)
+
+	if t.config.AWSProfile != "" {
+		opts = append(opts, awsconfig.WithSharedConfigProfile(t.config.AWSProfile))
+	}
+
+	cfg, err := awsconfig.LoadDefaultConfig(context.Background(), opts...)
+	if err != nil {
+		return fmt.Errorf("unable to load default config: %w", err)
+	}
+
+	if t.config.AWSRoleARN != "" {
+		// See https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials/stscreds#hdr-Assume_Role
+		// Create the credentials from AssumeRoleProvider to assume the role
+		// identified by AWSRoleARN, and use them in the loaded config
+		stsSvc := sts.NewFromConfig(cfg)
+		creds := stscreds.NewAssumeRoleProvider(stsSvc, t.config.AWSRoleARN)
+		cfg.Credentials = aws.NewCredentialsCache(creds)
+	}
+
+	t.awsCfgMut.Lock()
```

Comment:
Curious to understand why is this mutex needed? Are we calling `t.Init` more than once from somewhere?